tests/conformance_dsl/model.rs

```diff
@@ -225,31 +237,14 @@ impl PartialEq<Element> for ModelValue {
                 }
                 true
             }
-            ModelValue::Struct(fields) => {
-                if other.ion_type() != IonType::Struct {
-                    false
-                } else {
-                    let struct_val = other.as_struct().unwrap();
-                    for (field, val) in struct_val.fields() {
-                        let denoted_field = field.text().unwrap(); // TODO: Symbol IDs
-                        let denoted_symtok = SymbolToken::Text(denoted_field.to_string());
-                        if let Some(expected_val) = fields.get(&denoted_symtok) {
-                            if expected_val != val {
-                                return false;
-                            }
-                        }
-                    }
-                    true
-                }
-            }
+            ModelValue::Struct(_) => panic!("should not be using element eq for struct"),
```

Comment:
What is it supposed to be using?

Reply:
I transitioned over to using a regular reader instead of the element API in order to access symbol IDs, so the handling of structs, lists, and s-expressions moved over to the LazyValue PartialEq. I cleaned this one up a bit and dropped a comment. Next time I touch this I'll probably get rid of the PartialEqs entirely in favor of something that can return a Result.